### PR TITLE
Remove usage of deprecated Twig stuff

### PR DESCRIPTION
**Pull request by Koc**

### CODE REVIEW

**Comment by rande** [comment]
Do you have a link with twig deprecated calls ?

**Comment by Koc** [comment]
http://twig.sensiolabs.org/doc/deprecated.html

**Comment by Soullivaneuh** [comment]
@rande https://github.com/twigphp/Twig/blob/71b4dc2572eb822552e394a254a7bdf16be15448/lib/Twig/Function/Method.php#L13

**Comment by Soullivaneuh** [comment]
@Koc Could you update twig deps? https://github.com/sonata-project/SonataIntlBundle/blob/b0d6748f36474fe6e2f2923507b77bdae4c38a8e/composer.json#L23

Should be `~1.12`.

File: Twig/Extension/DateTimeExtension.php [hunk 1]
```diff
@@ -27,31 +27,12 @@ public function __construct(DateTimeHelper $helper)
         $this->helper = $helper;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function initRuntime(\Twig_Environment $environment)
-    {
-        $this->environment = $environment;
```
Comment (Soullivaneuh) [inline Twig/Extension/DateTimeExtension.php]: Why removing this?

File: Twig/Extension/LocaleExtension.php [hunk 2]
```diff
@@ -27,36 +27,17 @@ public function __construct(LocaleHelper $helper)
         $this->helper = $helper;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function initRuntime(\Twig_Environment $environment)
-    {
-        $this->environment = $environment;
```
Comment (Soullivaneuh) [inline Twig/Extension/LocaleExtension.php]: Why removing this?

File: Twig/Extension/DateTimeExtension.php (hunk 1 above)
Comment (Koc) [inline Twig/Extension/DateTimeExtension.php]: this property not defined in class and unused

File: Twig/Extension/LocaleExtension.php (hunk 2 above)
Comment (Koc) [inline Twig/Extension/LocaleExtension.php]: same reason

**Comment by Koc** [comment]
@Soullivaneuh updated twig deps

File: Templating/Helper/BaseHelper.php [hunk 3]
```diff
@@ -69,7 +69,7 @@ protected function fixCharset($string)
      */
     public static function getUCIDataVersion()
     {
-        trigger_error('The '.__METHOD__.' is deprecated since 2.2 and will be removed on 3.0. Use '.__CLASS__.'::getICUDataVersion instead.', E_USER_DEPRECATED);
+        @trigger_error('The '.__METHOD__.' is deprecated since 2.2 and will be removed on 3.0. Use '.__CLASS__.'::getICUDataVersion instead.', E_USER_DEPRECATED);
```
Comment (Soullivaneuh) [inline Templating/Helper/BaseHelper.php]: Please do not change code note related to the PR.

Submit another PR instead.

File: Templating/Helper/BaseHelper.php [hunk 4]
```diff
@@ -92,7 +92,7 @@ public static function getICUDataVersion()
 
         $info = explode("\n", $content);
 
-        if ('cli' == php_sapi_name()) {
+        if ('cli' === php_sapi_name()) {
```
Comment (Soullivaneuh) [inline Templating/Helper/BaseHelper.php]: Please do not change code note related to the PR.

Submit another PR instead.

File: Templating/Helper/BaseHelper.php [hunk 5]
```diff
@@ -113,8 +113,6 @@ public static function getICUDataVersion()
                 }
             }
         }
-
-        return;
```
Comment (Soullivaneuh) [inline Templating/Helper/BaseHelper.php]: Please do not change code note related to the PR.

Submit another PR instead.

File: Twig/Extension/DateTimeExtension.php (hunk 1 above)
Comment (Soullivaneuh) [inline Twig/Extension/DateTimeExtension.php]: :+1:

**Comment by Koc** [comment]
@Soullivaneuh I will remove unrelated changes today

**Comment by Soullivaneuh** [comment]
Ok, ping me when it's done.

**Comment by Koc** [comment]
@Soullivaneuh done

**Comment by Soullivaneuh** [comment]
Thank you @Koc.

**Comment by Koc** [comment]
@Soullivaneuh can you tag release, please?

**Comment by Soullivaneuh** [comment]
@Koc Just done: https://github.com/sonata-project/SonataIntlBundle/releases/tag/2.2.3
